client/src/pages/Join.js

```diff
@@ -129,7 +134,7 @@ function Join() {
           </Typography>
           <TextField
             value={matchId}
-            onChange={onTextChange}
+            onChange={(e) => onTextChange(e)}
```

Comment:
This change shouldn't be necessary, you should be able to just pass the function directly